pkg/metadata/inventories_collector.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package metadata
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery"
+	checkCollector "github.com/DataDog/datadog-agent/pkg/collector"
+	"github.com/DataDog/datadog-agent/pkg/metadata/inventories"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// InventoriesCollector fills and sends the inventory metadata payload
+type InventoriesCollector struct{}
+
+// Send collects the data needed and submits the payload
+func (hp *InventoriesCollector) Send(s *serializer.Serializer) error {
```

Comment:
I had a PR that does this same thing but also adds the non-regular sends: https://github.com/DataDog/datadog-agent/pull/4240

Sorry you had to reimplement it, I thought it was merged 🙏